src/dfx/src/lib/nns/install_nns.rs

```diff
@@ -1,32 +1,485 @@
+/// # Notes:
+///   - Set DFX_IC_NNS_INIT_PATH=<path to binary> to use a different &binary for local development
+///   - This won't work with an HSM, because the agent holds a session open
+///   - The provider_url is what the agent connects to, and forwards to the replica.
+/// # Prerequisites
+///   - There must be no canisters already present in the dfx server.
+///   - The dfx server must be running as subnet type system; this is set in the local network setting in dfx.json and
+///     will normally be different from the production network type, which will most
+///     likely be "application".
+/// # Errors
+/// This will return an error if:
+/// - Any of the steps failed to complete.
+///
+/// # Panics
+/// Ideally this should never panic and always return an error on error; while this code is in development reality may differ.
 #[context("Failed to install nns components.")]
 pub async fn install_nns(
-    _agent: &Agent,
+    env: &dyn Environment,
+    agent: &Agent,
     _provider_url: &str,
     ic_nns_init_path: &Path,
-    _replicated_state_dir: &Path,
-) -> DfxResult {
-    // Notes:
-    //   - Set DFX_IC_NNS_INIT_PATH=<path to binary> to use a different binary for local development
-    //   - This won't work with an HSM, because the agent holds a session open
-    //   - The provider_url is what the agent connects to, and forwards to the replica.
+    replicated_state_dir: &Path,
+) -> anyhow::Result<()> {
+    println!("Checking out the environment...");
+    // Check out the environment.
+    verify_local_replica_type_is_system()?;
+    let subnet_id = get_local_subnet_id(replicated_state_dir)?;
+    let nns_url = get_replica_url(env)?;
+
+    // Install the core backend wasm canisters
+    download_nns_wasms(env).await?;
+    let ic_nns_init_opts = IcNnsInitOpts {
+        wasm_dir: nns_wasm_dir(env),
+        nns_url: nns_url.to_string(),
+        test_accounts: Some(TEST_ACCOUNT.to_string()),
+        sns_subnets: Some(subnet_id.clone()),
+    };
+    ic_nns_init(ic_nns_init_path, &ic_nns_init_opts).await?;
+    let mut canister_id_store = CanisterIdStore::for_env(env)?;
+    canister_id_store.add("nns-registry", "rwlgt-iiaaa-aaaaa-aaaaa-cai")?;
+    canister_id_store.add("nns-governance", "rrkah-fqaaa-aaaaa-aaaaq-cai")?;
+    canister_id_store.add("nns-ledger", "ryjl3-tyaaa-aaaaa-aaaba-cai")?;
+    canister_id_store.add("nns-root", "r7inp-6aaaa-aaaaa-aaabq-cai")?;
+    canister_id_store.add("nns-cycles-minting", "rkp4c-7iaaa-aaaaa-aaaca-cai")?;
+    canister_id_store.add("nns-lifeline", "rno2w-sqaaa-aaaaa-aaacq-cai")?;
+    canister_id_store.add("nns-genesis-token", "renrk-eyaaa-aaaaa-aaada-cai")?;
+    // These two canisters are created but we create other canisters later for the same role.
+    // canister_id_store.add("nns-identity", "rdmx6-jaaaa-aaaaa-aaadq-cai")?;
+    // canister_id_store.add("nns-ui", "qoctq-giaaa-aaaaa-aaaea-cai");
+    canister_id_store.add("nns-sns-wasm", "qvhpv-4qaaa-aaaaa-aaagq-cai")?;
```

Comment:
Per the description at https://www.notion.so/SNS-developer-workflow-866b91041161497ebf6f58828451d70e:
> - unrelated to project

I'd expect this to happen in `dfx nns import` (which will put them in `remote` canister definitions in dfx.json, not canister_ids.json)

Reply:
The canister id store is always stored with the project.  It's either in
- <project dir>/canister_ids.json (for persistent ids)
- <project dir>/.dfx/<network>/canister_ids.json (for ephemeral ids)
`dfx import nns` will actually add `remote` definitions to dfx.json, rather than altering canister_ids.json.